wpiutil/src/main/native/cpp/SafeThread.cpp

```diff
@@ -2,10 +2,30 @@
 // Open Source Software; you can modify and/or share it under the terms of
 // the WPILib BSD license file in the root directory of this project.
 
+#include <atomic>
+
 #include "wpi/SafeThread.h"
 
 using namespace wpi;
 
+// thread start/stop notifications for bindings that need to set up
+// per-thread state
+
+static void* DefaultOnThreadStart() {
+  return nullptr;
+}
+static void DefaultOnThreadEnd(void*) {}
+
+static std::atomic<void* (*)()> gOnSafeThreadStart{DefaultOnThreadStart};
+static std::atomic<void (*)(void*)> gOnSafeThreadEnd{DefaultOnThreadEnd};
+
+namespace wpi::impl {
+void SetSafeThreadNotifers(void* (*OnStart)(), void (*OnEnd)(void*)) {
```

Comment:
Shouldn't there be a declaration in a header somewhere?

Reply:
There are a bunch of other RobotPy-specific functions that also aren't in headers. Probably should document them. They aren't in headers to ensure that users don't accidentally use them.

Reply:
These callbacks are probably safe to use anyway, right? I don't see users really being able to shoot themselves with this.

Reply:
Also, given they take C functors rather than std::function, these can be C exports (prefixed with WPI_) instead of C++ ones.